js/foundation.offcanvas.js

```diff
@@ -300,9 +380,15 @@ class OffCanvas extends Plugin {
     this.$triggers.attr('aria-expanded', 'false');
 
     if (this.options.trapFocus === true) {
-      this.$element.siblings('[data-off-canvas-content]').removeAttr('tabindex');
+      this.$content.removeAttr('tabindex');
       Keyboard.releaseFocus(this.$element);
     }
+
+    // Listen to transitionEnd and add class when done.
+    this.$element.one(Foundation.transitionend(this.$element), function(e) {
```

Comment:
done now in https://github.com/zurb/foundation-sites/pull/10003/commits/5aa588dcfc831c87b9262987e51e5af2d03f3e3f